acquia_lift.install

```diff
@@ -639,6 +638,186 @@ function acquia_lift_update_7017() {
+    variable_set('acquia_lift_legacy_agents', $legacy_agents);
+    return;
+  }
+
+  // Now convert anything that can be converted.
+  foreach ($convertibles as $original_agent_name => $details) {
+    // Create a new acquia_lift_target agent which will delegate to this one.
+    $agent = new stdClass();
+    $agent->label = 'Parent personalization for ' . $details[0]['agent_label'];
+    $agent->plugin = 'acquia_lift_target';
+    $agent->data = array();
+    $agent->machine_name = personalize_generate_machine_name($original_agent_name . '-parent', 'personalize_agent_machine_name_exists');
+    $nested_agent = personalize_agent_load($original_agent_name);
+    try {
+      $agent = personalize_agent_save($agent);
+      // Set the status of our new agent to that of the original.
+      $nested_agent_status = personalize_agent_get_status($original_agent_name);
+      personalize_agent_set_status($agent->machine_name, $nested_agent_status);
+      $agent->started = $nested_agent->started;
+      $agent = personalize_agent_save($agent);
+      // Move goals over to the new agent.
+      $goals = personalize_goal_load_by_conditions(array('agent' => $original_agent_name));
+      foreach ($goals as $goal) {
+        db_update('personalize_campaign_goals')
+          ->condition('id', $goal->id)
+          ->fields(array('agent' => $agent->machine_name))
+          ->execute();
+      }
+      // Remove the "decisions" info from the data property of the original
+      // agent as this no longer serves any function.
+      unset($nested_agent->data['decisions']);
+      personalize_agent_save($nested_agent);
+
+      $nested_osid = NULL;
+      // If there are mulitple option sets, they constitute one decision.
+      foreach ($details as $i => $option_set_details) {
+        $nested_os = personalize_option_set_load($option_set_details['osid']);
+        if ($i == 0) {
+          $nested_osid = $nested_os->osid;
+        }
+        $main_os = new stdClass();
+        $main_os->agent = $agent->machine_name;
+        $main_os->data = $nested_os->data;
+        $main_os->label = $nested_os->label;
+        $main_os->options = $nested_os->options;
+        $main_os->plugin = $nested_os->plugin;
+        $main_os->executor = $nested_os->executor;
+        $main_os->winner = $nested_os->winner;
+        $main_os->stateful = $nested_os->stateful;
+        $main_os->preview_link = $nested_os->preview_link;
+        $main_os->decision_name = $option_set_details['decision_name'];
+        $main_os->targeting = array(
+          ACQUIA_LIFT_TARGETING_EVERYONE_ELSE => array(
+            'osid' => $nested_osid
+          )
+        );
+        personalize_option_set_save($main_os);
+        // Convert the first of the original option sets to a nested option set
+        // and delete the others.
+        if ($i == 0) {
```

Comment:
Because for each one we've already created a new option set on the parent agent. Each of them will reference the same nested test in their targeting property.